ui/cypress/integration/sidebar.spec.js

```diff
@@ -0,0 +1,78 @@
+beforeEach(() => {
+  cy.setupMocks();
+  cy.login();
+});
+
+describe('Sidebar', () => {
+  beforeEach(() => {
+    cy.window()
+      .its('localStorage')
+      .then((store) => store.setItem('sidebar_expanded', true));
+    cy.visit('/');
+  });
+
+  // TODO: Remove the interaction from the E2E test as much as possible
+  it('brings me to the monitoring page', () => {
+    cy.window()
+      .its('__history__')
+      .then((history) => {
+        cy.stub(history, 'push').as('historyPush');
+      });
+
+    cy.get('[data-cy="sidebar_item_monitoring"]').click();
+
+    cy.get('@historyPush').should('be.calledWith', '/');
+  });
+
+  it('brings me to the node page', () => {
+    cy.window()
+      .its('__history__')
+      .then((history) => {
+        cy.stub(history, 'push').as('historyPush');
+      });
+
+    cy.get('[data-cy="sidebar_item_nodes"]').click();
+
+    cy.get('@historyPush').should('be.calledWith', '/nodes');
+  });
+
+  it('brings me to the volume page', () => {
+    cy.window()
+      .its('__history__')
+      .then((history) => {
+        cy.stub(history, 'push').as('historyPush');
+      });
+
+    cy.get('[data-cy="sidebar_item_volumes"]').click();
+
+    cy.get('@historyPush').should('be.calledWith', '/volumes');
+  });
+
+  it('brings me to the environment page', () => {
+    cy.window()
+      .its('__history__')
+      .then((history) => {
+        cy.stub(history, 'push').as('historyPush');
+      });
+
+    cy.get('[data-cy="sidebar_item_environments"]').click();
+
+    cy.get('@historyPush').should('be.calledWith', '/environments');
+  });
+
+  it('can expand', () => {
+    let isSidebarExpanded;
+
+    cy.get('.sc-sidebar div:first .sc-button').click();
+    cy.window()
+      .its('localStorage')
+      .then((store) => {
+        isSidebarExpanded = store.getItem('sidebar_expanded');
+      })
+      .then(() => {
+        // Cypress commands are asynchronous, so you cannot check a property value before the Cypress commands ran.
+        // use cy.then() callback to check the value.
+        expect(isSidebarExpanded).to.equal('false');
+      });
+  });
```

Comment:
If this test checks that the sidebar `can expand`, I would expect `isSidebarExpanded` to be true, at the end.

Now, this test should not depend on "the default state of the app" (here, it's expanded by default), otherwise the test will fail once we change this default, while the feature still works.

Here's how I would rewrite those tests:

```javascript
  it('is expanded by default', () => {
    // assuming we have set the proper attributes
    // 'data-cy' and 'data-cy-state' in `sidebarConfig`
    cy.get('[data-cy="sidebar"]').should(
      'have.attr',
      'data-cy-state',
      'expanded',
    );

    cy.window()
      .its('localStorage')
      .invoke('getItem', 'sidebar_expanded')
      .should('equal', 'true');
  });

  it('can be collapsed', () => {
    cy.window()
      .its('localStorage')
      .invoke('setItem', 'sidebar_expanded', 'true');

    cy.get('[data-cy="sidebar"] > :first').click();

    cy.get('[data-cy="sidebar"]').should(
      'have.attr',
      'data-cy-state',
      'collapsed',
    );
    cy.window()
      .its('localStorage')
      .invoke('getItem', 'sidebar_expanded')
      .should('equal', 'false');
  });

  it('can be expanded', () => {
    cy.window()
      .its('localStorage')
      .invoke('setItem', 'sidebar_expanded', 'false');

    cy.get('[data-cy="sidebar"] > :first').click();

    cy.get('[data-cy="sidebar"]').should(
      'have.attr',
      'data-cy-state',
      'expanded',
    );
    cy.window()
      .its('localStorage')
      .invoke('getItem', 'sidebar_expanded')
      .should('equal', 'true');
  });

Reply:
In the third case, we may not be able to get the default sidebar_expanded item in` localStorage` though.
Because if do nothing to the sidebar, we don't set it in localStorage. the default state would be in the redux store.

Reply:
I guess you mean "in the first case", i.e. `it('is expanded by default')`?
I did see this when running the app with a clear localStorage (nothing set for `sidebar_expanded`), but somehow when running using Cypress it was set (and checking that it wasn't set failed...). Any idea why that would be the case?